### ISSUE TITLE
[Bug] there is an obvious flicker when the defaultSizes contains zero on first render

### ISSUE THREAD

**Issue by percy507** [johnwalley/allotment#537]
See the gif, the defaultSizes is `[0, 10000]`. I refresh page three times. The flicker occurs twice.

![2023-01-29 16 28 05](https://user-images.githubusercontent.com/18365078/215315654-2a1d21fc-556c-47a2-9aaf-4ae85326743a.gif)

**Comment by percy507** [comment]
Ahhh, I notice there is a PR (https://github.com/johnwalley/allotment/pull/526) which can solve this problem.

**Comment by johnwalley** [comment]
Hi @percy507. Thanks for submitting this issue. I'll get onto #526 and hopefully that will resolve this.

**Comment by johnwalley** [comment]
#526 has been merged and released as v1.18.0. Let me know if it helps 👍

**Comment by percy507** [comment]
Hi @johnwalley,

> See the gif, the defaultSizes is `[0, 10000]`. I refresh page three times. The flicker occurs twice.

Well, after some test, the #526 can resolve the problem what I found in storybook. 

But unfortunately, the same problem happens in my personal project. I made a demo, it can reproduce the flicker.  https://codesandbox.io/s/thirsty-wind-bzqg9h?file=/src/App.tsx

After some test and debug, I found some unwanted behaviors.

- After initializing the `Allotment` component, the `onChange` method is called which is not triggered manually.
- When the `visible` of `Pane` is false, the init size of the pane should be `0` before it mounted. But during my debug, I found it's size first init as its `minSize`, then cause the `visible` is false, re-layout to the right size. (I guess this is why the flicker occurs)
  - The related code might be here. https://github.com/johnwalley/allotment/blob/main/src/split-view/split-view.ts#L1022
